### DISCUSSION TITLE
New major release V2

### DISCUSSION THREAD

**Discussion by albaintor** [albaintor/homeassistant_electrolux_status#66] [Announcements]
Many thanks to @kingy444 who realized all of these major improvements 
Merged from https://github.com/albaintor/homeassistant_electrolux_status/pull/61

- Rename across the board to Electrolux Status for consistency and clear identification
- Complete code refactor to conform to a core integration including applying pylint, ruff and typehinting throughout
- Reduce code footprint overall, splitting into smaller modules and reusing code snippets
- Use HomeAssistant aiohttp session rather than spinning up our own
- Implement Store for token within the session, allowing infinite restarts and refresh of token rather than re-authentication every 12 hours
- Improve logging for consistency
- Nested elements no longer use root and element seperately, now using root/element as key to remove clashes in device types
- Fix recursive lookup for nested elements (Fridge / Freezer)
- Implement dataclass model for ElectroluxDevice to improve code quality
- Ability to override capability on ElectroluxDevice based on the model / type
- Move catalog to its own module(s) and out of const.py
- Enhance config flow
- add ability to subscribe/unsubscribe to alerts
- add change password / language
- only one instance per account
- Add diagnostics support
- Add sample directory (suggest expanding this with more device types (diagnostics)
- Add Translations for supported languages. (These need to be regenerated when strings change)
- Add support for constant and alert types
- cloud_push over cloud_poll
- Minimise default entities to most useful.
- bump upstream dependancy
- 'Sister' entities capabiity for devices that have their status linked to another (ice maker on/off control and state are different entities, this allows a toggle rather than a button)
- Mapping of values from API to user state
- Mapping of string values to running states for true binary_sensor to work
- Button creation is more dynamic and controllable
- Device type can be forced via the catalog type (simply setting device_class in catalog can chnage the entity type created)

<hr /><em>This discussion was created from the release <a href='https://github.com/albaintor/homeassistant_electrolux_status/releases/tag/v2.0.0'>New major release V2</a>.</em>

**Comment by wgumaa** [comment]
Thanks @albaintor for this release. I see a lot of effort has gone into streamlining the integration. However, I have noticed some values and entities that were available for my PerfectCare washing machine have now disappeared. Examples: 

1. analogspinspeed - Spin speed
2. programuid - Washing programme - Cottons, delicates etc. 
3. analogtemperature - Temperature

To name a few. Am I missing something somewhere?

**Reply by kingy444** [reply]
Can you double check they haven't been renamed (or its type changed from say a number to a sensor). 
They might have been recreated under a new entityID and are under a different entity name now.

You could delete the integration, restart HA, re-configure the integration and they will likely get the old name back.

Otherwise can you open an issue and share the diagnostic file the integration generates

**Reply by wgumaa** [reply]
@kingy444 They are definitely not there. I did reinstall the integration to get rid of the duplicate names. Will log an issue.

**Comment by Imperial-Guard** [comment]
First off all @albaintor and @kingy444 thanks for all your effort into this. 

Since the update to the latest version the timer "aeg_mijn_oven_timetoend" is not reaching 0 and is stuck on 1. 
Any thoughts on this or should I create a seperate issue for this? It working before so it looks like that's related to some rewrite of the code. 

Product type - Steam Oven KSK998230T

Thanks :)

**Reply by kingy444** [reply]
can you open an issue and share the diagnostic file
